CodeForPoznan/pah-fm
# Align migrations with master branch

### Why
We were not concerned with `master` for few sprints which caused a bit of a drift from current state of database in production environment. I'm proposing these changes as a way to align it safely and automatically with database schema/migrations in `develop`.

### How to test it:
- clean your docker-compose environment including volumes
- checkout to master branch
- start project, generate some random data
- stop the project, checkout `package-lock.json`
- fetch this branch, checkout to it and start the project
- run migrations

Or use this handy one-liner :)
```
make stop && \
docker-compose down --volumes && \
git stash && \
git checkout -- frontend/package-lock.json && \
git checkout upstream/master && \ 
make start && \
sleep 20 && \
make populate-database && \
make stop && \
git checkout -- frontend/package-lock.json && \
git checkout align-migrations-with-master-branch
```

now watch closely the logs - migrations will be applied on start
```
make start logs
```

The result should be a neat output of applied migrations without any errors or exceptions.
